situp-core/integrationTest.gradle

```diff
@@ -0,0 +1,114 @@
+            srcDir file('src/integrationTest/java')
+        }
+        resources.srcDir file('src/integrationTest/resources')
+    }
+}
+
+dependencies {
+    integrationTestCompile("junit:junit:${versionMap.junit}")
+    integrationTestCompile project(':situp-plugins:elasticsearch')
+    integrationTestImplementation 'io.opentelemetry:opentelemetry-proto:0.6.0'
+    integrationTestImplementation 'com.google.protobuf:protobuf-java-util:3.13.0'
+    integrationTestImplementation ("com.linecorp.armeria:armeria:1.0.0")
+    integrationTestImplementation "org.elasticsearch.client:elasticsearch-rest-high-level-client:7.8.1"
+}
+
+/**
+ * Situp Docker tasks
+ */
+task createSitupDockerFile(type: Dockerfile) {
+    dependsOn jar
+    destFile = project.file('build/docker/Dockerfile')
+    from("adoptopenjdk/openjdk14:jre-14.0.1_7-alpine")
+    exposePort(21890)
+    workingDir("/app")
+    copyFile("build/libs/${jar.archiveName}", "/app/situp.jar")
+    copyFile("src/integrationTest/resources/pipeline.yml", "/app/pipeline.yml")
+    defaultCommand("java", "-jar", "situp.jar", "/app/pipeline.yml")
+}
+
+task buildSitupDockerImage(type: DockerBuildImage) {
+    dependsOn createSitupDockerFile
+    inputDir  = file(".")
+    dockerFile  = file("build/docker/Dockerfile")
+    images.add("integ-test-pipeline-image")
+}
+
+task createSitupDockerContainer(type: DockerCreateContainer) {
+    dependsOn buildSitupDockerImage
+    hostConfig.portBindings = ['21890:21890']
+    targetImageId buildSitupDockerImage.getImageId()
+}
+
+task startSitupDockerContainer(type: DockerStartContainer) {
+    dependsOn createSitupDockerContainer
+    targetContainerId createSitupDockerContainer.getContainerId()
+    doLast {
+        sleep(10*1000)
+    }
+}
+
+task stopSitupDockerContainer(type: DockerStopContainer) {
+    targetContainerId createSitupDockerContainer.getContainerId()
+}
+
+/**
+ * ODFE Docker tasks
+ */
+task createOdfeDockerContainer(type: DockerCreateContainer) {
+    targetImageId 'amazon/opendistro-for-elasticsearch:1.9.0'
+    hostConfig.portBindings = ['9200:9200', '9600:9600']
+    hostConfig.autoRemove = true
+    envVars = ['discovery.type':'single-node']
+}
+
+task startOdfeDockerContainer(type: DockerStartContainer) {
+    dependsOn createOdfeDockerContainer
+    targetContainerId createOdfeDockerContainer.getContainerId()
+
+    doLast {
+        sleep(90*1000)
+    }
+}
+
+task stopOdfeDockerContainer(type: DockerStopContainer) {
+    targetContainerId createOdfeDockerContainer.getContainerId()
+}
+
+/**
+ * End to end test. Spins up ODFE and situp docker containers, then runs the integ test
+ * Stops the docker containers when finished
+ */
+task endToEndTest(type: Test) {
+    dependsOn build
+    dependsOn startOdfeDockerContainer
+    dependsOn startSitupDockerContainer
+    startSitupDockerContainer.mustRunAfter 'startOdfeDockerContainer'
+
+    description = 'Runs the integration tests.'
```

Comment:
nit: call it situp or situp-core integration test, will make debug easier.

Reply:
Sure